debs/SPECS/4.0.0/wazuh-manager/debian/preinst

```diff
@@ -0,0 +1,136 @@
+#!/bin/sh
+# preinst script for Wazuh
+
+set -e
+
+# configuration variables
+DIR="/var/ossec"
+WAZUH_TMP_DIR="${DIR}/packages_files/manager_config_files"
+VERSION="$2"
+MAJOR=$(echo "$VERSION" | cut -dv -f2 | cut -d. -f1)
+
+# environment configuration
+if [ ! -d ${WAZUH_TMP_DIR} ]; then
+    mkdir -p ${WAZUH_TMP_DIR}
+else
+    rm -rf ${WAZUH_TMP_DIR}
+    mkdir -p ${WAZUH_TMP_DIR}
+fi
+
+case "$1" in
+    install|upgrade)
+
+        if [ ! -z "$2" ] && [ ! -f ${DIR}/etc/ossec.conf ] ; then
+            touch ${WAZUH_TMP_DIR}/create_conf
+        fi
+        # Force deleting the old ossec-init.conf and link
+        rm -f /etc/ossec-init.conf
+
+        if [ -z "$2" ]; then
+          if [ -f ${DIR}/etc/ossec.conf ]; then
+            cp  ${DIR}/etc/ossec.conf  ${DIR}/etc/ossec.conf.deborig
+            chmod 0600 ${DIR}/etc/ossec.conf.deborig
+            chown root:root ${DIR}/etc/ossec.conf.deborig
+
+            echo "====================================================================================="
+            echo "= Backup from your ossec.conf has been created at ${DIR}/etc/ossec.conf.deborig ="
+            echo "= Please verify your ossec.conf configuration at ${DIR}/etc/ossec.conf          ="
+            echo "====================================================================================="
+          fi
+        fi
+        # Stop API service if it's running
+        if ps aux | grep "${DIR}/api/app.js" | grep -v grep > /dev/null 2>&1; then
+           touch ${WAZUH_TMP_DIR}/wazuh-api.restart
+           if [ -n "$(ps -e | grep -E ^\ *1\ .*systemd$)" ]; then
+              systemctl --quiet stop wazuh-api.service
+           else
+              service wazuh-api stop
+           fi
+        fi
```

Comment:
Now, both manager and api services must be stopped before upgrading. Maybe it would be interesting to move this to the `prerm` script but it may affect the upgrades from 3.x.

Anyway, make sure both services are stopped.